### PR TITLE
Updates for v1.22.0 release

### PR DESCRIPTION
**Pull request by nurmukhametov**
This PR adds docs and release notes updates, bumps version, add release dockers.

### CODE REVIEW

File: docker/v1.22.0/ubuntu18.04/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,122 @@
+#
+#  Copyright (c) 2017-2023, Intel Corporation
+#
+#  SPDX-License-Identifier: BSD-3-Clause
+
+ARG LLVM_VERSION=16.0
+
+FROM ubuntu:18.04 AS llvm_build_step1
+LABEL maintainer="Dmitry Babokin <dmitry.y.babokin@intel.com>"
```
Comment (dbabokin) [inline docker/v1.22.0/ubuntu18.04/Dockerfile:9]:
```suggestion
LABEL maintainer="Aleksei Nurmukhametov <aleksei.nurmukhametov@intel.com>"
```

File: docs/ReleaseNotes.txt [hunk 2]
```diff
@@ -1,3 +1,74 @@
...
+
+ISPC release with template operators support; improved debugging experience of ISPC code on Windows; multiple
+stability and performance fixes and more. The release is based on patched LLVM 16.0.6.
+
+ISPC distribution changes:
+
+ - ISPC binaries were compiled with LTO by Clang/LLVM toolchain on all supported platforms and architectures using
+   superbuild. ISPC binaries got faster a few percent in average.
+ - Examples were excluded from ISPC archives. They are placed alongside as separate archives ispc-examples-v1.22.0.zip
+   and ispc-examples-v1.22.0.tar.gz.
```
Comment (dbabokin) [inline docs/ReleaseNotes.txt:11]:
```suggestion
 - Examples were excluded from ISPC archives. They are placed alongside as separate archives `ispc-examples-v1.22.0.zip`
   and `ispc-examples-v1.22.0.tar.gz`.
```

File: docs/ReleaseNotes.txt [hunk 3]
```diff
@@ -1,3 +1,74 @@
...
+Language changes:
+
+ - Added support for template operators.
+ - Revised the usage of function specifiers with templates. For more details please refer to Function Templates
+   section of documentation.
+
+Infrastructure changes:
+
+ - Release built with LTO (expect aarch64 Linux).
+ - Supported building ISPC with LLVM 17.
```
Comment (dbabokin) [inline docs/ReleaseNotes.txt:22]: Would be good to give a reason why the release is not with LLVM 17 - i.e. GPU support is incomplete, there are known regressions, was not tested throughly enough, etc.

File: docs/ReleaseNotes.txt [hunk 4]
```diff
@@ -1,3 +1,74 @@
...
+
+Language changes:
+
+ - Added support for template operators.
+ - Revised the usage of function specifiers with templates. For more details please refer to Function Templates
+   section of documentation.
+
+Infrastructure changes:
+
+ - Release built with LTO (expect aarch64 Linux).
```
Comment (dbabokin) [inline docs/ReleaseNotes.txt:21]:
```suggestion
 - Release built with LTO (expect `aarch64` Linux).
```

**Review by dbabokin — COMMENTED** [review]
I think you forgot to update version in `snap/snapcraft.yaml`

The rest seems ok.

File: docs/ReleaseNotes.txt (hunk 4 above)
Comment (pthariensflame) [reply docs/ReleaseNotes.txt:21]: Also:
```suggestion
 - Release built with LTO (except aarch64 Linux).
```

**Comment by nurmukhametov** [comment]
> snap/snapcraft.yaml

Since PR https://github.com/ispc/ispc/pull/2643, it is not need anymore. Version is parameter of CI job.

**Pull request merged**